xds/internal/client/xds.go

```diff
@@ -49,25 +49,25 @@ const transportSocketName = "envoy.transport_sockets.tls"
 // UnmarshalListener processes resources received in an LDS response, validates
 // them, and transforms them into a native struct which contains only fields we
 // are interested in.
-func UnmarshalListener(resources []*anypb.Any, logger *grpclog.PrefixLogger) (map[string]ListenerUpdate, error) {
+func UnmarshalListener(version string, resources []*anypb.Any, logger *grpclog.PrefixLogger) (map[string]ListenerUpdate, UpdateMetadata, error) {
```

Comment:
Why do we need the `version` here? Just to populate the metadata?
And we can't directly pass the full `DiscoveryResponse` because we have to support both `v2` and `v3`?
Just wondering if we can do any better ...

Reply:
`version` will be added to metadata in the next PR.
I didn't move this into that PR because merging changes on the same line causes troubles.

And yes, because we need to support v2 and v3, so we only pass the `any`s.